### PR TITLE
Adding updated analysis notebooks, relevant code files, and figures

### PR DESCRIPTION
**Pull request by kassabry**
I added all the updated notebooks and code files that helped create the new figures. I did not want to mess with the other branches so I created a new one that also adds the analysis notebook found in the other branch, so that one is safe to delete. It should have everything I modified and the new figures that were created. 

exp3/
- Added all the boxplot figures from experiment 3 that were generated.

exp4/
- Added different versions of the same updated figure for experiment 4, that compared CBWS hits found against docking and random.
- Added a new code file that automatically plots the molecular images onto the figure mentioned above.

exp_0/figs/
- Added two figures corresponding to batch size and hits ratio for the hyperparameter sweep generated by the Analyze Experiment 0 notebook. 

exp_1/figs
- Added the updated boxplots for the hyperparameter sweep generated by the Analyze Experiment 1 notebook.

./
- Fixed a broken plotting method in the analyze_helper_exp3.py method, also changed some indexing that was causing problems
- Re-generated figures and fixed data paths in the Analyze Experiments 0,1,2 and 4 notebooks
- Fixed method in stat_analysis.py because I could not recreate the rainbow coloring found in original image
- Added PstP_Screening_Data_Analysis notebook that does some high-level analysis of the results from the PstP experiment such as number of actives by CBWS/Docking, testing PAINS filters, and showing the data split procedures

### CODE REVIEW

**Comment by kassabry** [comment]
Both Analysis Notebook 4: PstP results and Analysis Notebook 4: One Round screening were updated with the correct data paths, and had figures updated to match our discussion. 

The Analysis Notebook 4: One Round screening had some code regenerated due to some wasted effort on recalculating the Tanimoto distance matrix several times. Random seed sampling was also added for diversity sampling to replicate the diversity sampling done in the paper. Directions were added to point to running a code file in chtc_runners/ to generate the selected.csv.gz files necessary for figure generation. 

run_all_ors_pstp.py was created in chtc_runners/ to run all of the generated data in Analysis Notebook 4: One Round Screening based on the training data. 

The experiment_ors_pstp_runner.py was also changed slightly to address some data path issues. 

The changes with the other analysis notebooks are related to metadata, and no changes were made, since the first commit.
